File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/SpringBootConfigurationSpringBootSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableSpringBootConfigurationSpringBootSettings.class)
public interface SpringBootConfigurationSpringBootSettings {
  List<OverrideParameter> overrideParameters();
}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/configurations/spring_boot/SpringBootConfigurationSpringBootSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.configurations.spring_boot

import com.fasterxml.jackson.annotation.JsonProperty

data class SpringBootConfigurationSpringBootSettings(

    @JsonProperty("Enable debug output")
    val enableDebugOutput: Boolean? = null,

    @JsonProperty("Hide banner")
    val hideBanner: Boolean? = null,

    @JsonProperty("Enable launch optimization")
    val enableLaunchOptimization: Boolean? = null,

    @JsonProperty("Enable JMX agent")
    val enableJmxAgent: Boolean? = null,

    @JsonProperty("Override parameters")
    val overrideParameters: List<OverrideParameter>? = null
)
